wrappers/java/src/main/java/org/hyperledger/indy/sdk/agent/Agent.java

```diff
@@ -0,0 +1,329 @@
+import java.util.concurrent.Future;
+
+import org.hyperledger.indy.sdk.LibSovrin;
+import org.hyperledger.indy.sdk.SovrinException;
+import org.hyperledger.indy.sdk.SovrinJava;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentAddIdentityResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentCloseConnectionResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentCloseListenerResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentConnectResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentListenResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentRemoveIdentityResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentSendResult;
+import org.hyperledger.indy.sdk.pool.Pool;
+import org.hyperledger.indy.sdk.wallet.Wallet;
+
+import com.sun.jna.Callback;
+
+/**
+ * agent.rs API
+ */
+public class Agent extends SovrinJava.API {
+
+	private Agent() {
+
+	}
+
+	/*
+	 * STATIC METHODS
+	 */
+
+	public static Future<AgentConnectResult> agentConnect(
+			Pool pool,
+			Wallet wallet,
+			String senderDid,
+			String receiverDid,
+			Callback messageCb) throws SovrinException {
+
+		final CompletableFuture<AgentConnectResult> future = new CompletableFuture<> ();
+
+		Callback connectionCb = new Callback() {
+
+			@SuppressWarnings("unused")
+			public void callback(int xcommand_handle, int err, int connection_handle) {
+
+				if (! checkCallback(future, xcommand_handle, err)) return;
+
+				Agent.Connection connection = new Agent.Connection(connection_handle);
+
+				AgentConnectResult result = new AgentConnectResult(connection);
+				future.complete(result);
+			}
+		};
+
+		int poolHandle = pool.getPoolHandle();
+		int walletHandle = wallet.getWalletHandle();
+
+		int result = LibSovrin.api.sovrin_agent_connect(
+				FIXED_COMMAND_HANDLE, 
+				poolHandle,
+				walletHandle, 
+				senderDid,
+				receiverDid,
+				connectionCb,
+				messageCb);
+
+		checkResult(result);
+
+		return future;
+	}
+
+	public static Future<AgentListenResult> agentListen(
+			String endpoint,
+			Callback connectionCb,
+			Callback messageCb) throws SovrinException {
+
+		final CompletableFuture<AgentListenResult> future = new CompletableFuture<> ();
+
+		Callback listenerCb = new Callback() {
+
+			@SuppressWarnings("unused")
+			public void callback(int xcommand_handle, int err, int listener_handle) {
+
+				if (! checkCallback(future, xcommand_handle, err)) return;
+
+				Agent.Listener connection = new Agent.Listener(listener_handle);
+
+				AgentListenResult result = new AgentListenResult(connection);
+				future.complete(result);
+			}
+		};
+
+		int result = LibSovrin.api.sovrin_agent_listen(
+				FIXED_COMMAND_HANDLE, 
+				endpoint,
+				listenerCb,
+				connectionCb,
+				messageCb);
+
+		checkResult(result);
+
+		return future;
+	}
+
+	public static Future<AgentAddIdentityResult> agentAddIdentity(
+			Agent.Listener listener,
+			Pool pool,
+			Wallet wallet,
+			String did,
+			Callback connectionCb,
```

Comment:
> Callback connectionCb,
> Callback messageCb

I don't understand why agentAddIdentity and agentRemoveIdentity require these params. It looks like copy-paste issue.